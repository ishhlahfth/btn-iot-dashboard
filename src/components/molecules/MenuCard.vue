<template>
  <div class="grid grid-flow-col gap-x-4 py-2 sm:p-2 menu-card">
    <img
      v-if="imageUrl"
      :src="imageUrl"
      alt="menu"
      class="w-20 h-20 sm:w-26 sm:h-26 object-cover rounded"
    />
    <div
      v-else
      class="w-20 h-20 sm:w-26 sm:h-26 border-2 border-dashed border-grey-4 rounded grid place-items-center text-small"
    >
      No Image
    </div>
    <div class="flex flex-col justify-between">
      <div class="grid gap-1 sm:gap-2">
        <div class="grid sm:gap-1">
          <div class="grid grid-flow-col gap-2 auto-cols-max place-items-center">
            <p class="font-medium">{{ name }}</p>
            <help-badge
              :label="availabilityStatus ? 'Available' : 'Unavailable'"
              :color="availabilityStatus ? 'positive' : 'negative'"
            />
          </div>
          <p class="text-small text-grey-3">{{ category }}</p>
        </div>
        <p class="text-small text-grey-2 mb-2 sm-truncate">
          {{ description }}
        </p>
      </div>
      <p class="text-small font-medium">{{ store.methods.convertToRp(price) }}</p>
    </div>
    <div class="hidden h-26 sm:grid grid-flow-col place-items-center gap-2">
      <help-toggle />
      <icon name="chevron-down" class="cursor-pointer" @click="variantOpened = !variantOpened" />
    </div>
  </div>

  <transition name="slide" appear>
    <template v-if="variantOpened">
      <div class="grid text-small divide-y divide-grey-4">
        <template v-for="(variant, i) in variants" :key="i">
          <div class="p-2">
            <p class="mb-2">{{ variant.variant_name }}</p>
            <div class="grid sm:grid-cols-4 text-grey-2">
              <template v-for="(variantItem, i) in variant.variant_items" :key="i">
                <help-checkbox v-if="variant.multiple_choice" :label="variantItem" disabled />
                <help-radio v-else :label="variantItem" disabled />
              </template>
            </div>
          </div>
        </template>
      </div>
    </template>
  </transition>
</template>

<script>
import { inject, ref } from 'vue';
import HelpBadge from '@/components/atoms/Badge.vue';
import HelpCheckbox from '@/components/atoms/Checkbox.vue';
import HelpRadio from '@/components/atoms/Radio.vue';
import HelpToggle from '@/components/atoms/Toggle.vue';
import Icon from '@/components/atoms/Icon.vue';

export default {
  name: 'MenuCard',
  components: {
    HelpBadge,
    HelpCheckbox,
    HelpRadio,
    HelpToggle,
    Icon,
  },
  props: {
    imageUrl: {
      type: String,
      default: '',
    },
    name: {
      type: String,
      default: 'Untitled',
    },
    category: {
      type: String,
      default: '-',
    },
    description: {
      type: String,
      default: '-',
    },
    price: {
      type: Number,
      default: 0,
    },
    availabilityStatus: {
      type: Boolean,
      default: true,
    },
    isActive: {
      type: Boolean,
      default: true,
    },
    variants: {
      type: Array,
      default: () => [],
    },
  },
  setup() {
    const store = inject('store');
    const variantOpened = ref(false);
    return { store, variantOpened };
  },
};
</script>

<style lang="scss" scoped>
.sm-truncate {
  @apply truncate;
  @media screen and (min-width: 640px) {
    overflow: initial;
    text-overflow: initial;
    white-space: initial;
  }
}
.menu-card {
  grid-template-columns: auto 1fr;
  @media screen and (min-width: 640px) {
    grid-template-columns: auto 1fr auto auto;
  }
}
.slide-enter-active,
.slide-leave-active {
  transition: all 0.2s ease;
}
.slide-enter-from,
.slide-leave-to {
  transform: translateY(-10px);
  opacity: 0;
}
</style>
