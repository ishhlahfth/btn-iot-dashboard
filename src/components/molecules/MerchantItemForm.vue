<template>
  <help-modal v-model="modal.sm" permanent>
    <confirmation
      title="Delete Photo Product"
      message="Are you sure you want to delete this photo product ?"
      bgColor="red-500"
      @close="modal.sm = false"
      @cancel="modal.sm = false"
      @confirm="confirmDelete"
    />
  </help-modal>
  <div v-if="flagVarianGroup">
    <varian-options @closeVarian="flagVarianGroup = false" @handleAddVarian="handleAddVarian" @handleResetVarian="handleResetVarian" @submitVarian="submitVarian" :flagEditVarian="flagEditVarian" :itemVarians="itemVarians" />
  </div>
  <div v-else-if="flagItemCatalog">
    <product-catalog @handleAddCatalog="handleAddCatalog" @handleEditCatalog="handleEditCatalog" @handleDeleteCatalog="handleDeleteCatalog" :itemCatalogs="itemCatalogs" @close="flagItemCatalog = false" />
  </div>
  <div v-else>
    <div class="inner-modal-fixed overflow-auto px-1">
      <div class="divide-y divide-grey-4">
        <div class="flex justify-between py-2">
          <span class="font-semibold text-heading4">ADD PRODUCT</span>
        </div>
        <p></p>
      </div>
      <div class="py-2 lg:py-5 grid gap-2">
        <span class="font-semibold text-heading5">PRODUCT IMAGES (0/5)</span>
        <div
          class="py-2 lg:py-3 h-20 lg:h-26 xl:h-32 px-2 lg:px-3 border border-grey-4 rounded grid grid-cols-5 gap-1 md:gap-2 lg:gap-2"
        >
          <div
            v-for="(productImage, i) in productImages"
            :key="i"
            class="bg-grey-4 flex items-center justify-center rounded-md lg:w-full lg:h-full"
            @mouseover="handleHover(i, true)"
            @mouseleave="handleHover(i, false)"
          >
            <help-thumbnail
              :key="i"
              use-bg
              class="hover:opacity-50"
              width="100%"
              height="100%"
              :src="
                productImage
                  ? productImage
                  : 'https://www.couvee.co.id/wp-content/uploads/2019/11/CF4566E9-0DC2-43F1-ABC9-F1BED1F0A9CE-768x768.jpg'
              "
            />
            <help-button
              :key="i"
              v-if="photoHover[`${i}`]"
              icon-only
              icon="trash"
              bg-color="red-600"
              class="absolute top-34"
              @click="handleDeletePhoto(productImage)"
            />
          </div>
          <div
            @click="$refs.photoProduct.click()"
            class="relative bg-gray-300 flex items-center justify-center rounded-md w-16 h-16 lg:h-20 lg:w-20 xl:w-26 xl:h-26 hover:bg-gray-400 cursor-pointer"
            :class="{
              hidden: productImages.length === 5,
            }"
          >
            <icon name="plus-circle" :size="32" color="#ffffff" />
            <input
              ref="photoProduct"
              type="file"
              accept="image/*"
              class="h-full w-full opacity-0 hidden"
              @input="handleChangeImg"
            />
          </div>
        </div>
      </div>
      <div class="grid gap-2">
        <span class="font-semibold text-heading5">PRODUCT DETAIL</span>
        <help-input
          type="text"
          label="Product Name"
          placeholder="Product Name here"
          left-icon="template-outline"
        />
        <help-input
          type="text"
          label="Product Description"
          placeholder="Product Description here"
          left-icon="description"
        />
        <help-input
          type="number"
          label="Product Price (Rp)"
          placeholder="Rp 1.000.000"
          left-icon="price"
        />
        <help-select
          label="Stock Status"
          :options="stocks"
          v-model="form.status"
          :position="['bottom', 'right']"
          left-icon="status-stock"
        />
        <help-input
          type="number"
          label="Minimum Order"
          placeholder="Product Minimum to Order"
          left-icon="cart-outline"
        />
        <help-select
          label="Product Category"
          :options="itemCategories"
          v-model="form.catalog_id"
          :position="['bottom', 'right']"
          left-icon="tag"
        />
        <template class="grid grid-cols-4 gap-2">
          <help-select
            class="col-span-3"
            label="Product Catalog"
            :options="itemCatalogs"
            v-model="form.productCatalog"
            :position="['bottom', 'right']"
            left-icon="catalog"
          />
          <div class="w-full outline-none grid">
            <label class="font-medium"></label>
            <div class="flex items-center justify-start pt-5 cursor-pointer">
              <help-button
                bg-color="blue-500"
                icon="plus-circle"
                class="h-full w-11/12 md:w-full"
                label="Catalog"
                @click="flagItemCatalog = true"
              />
            </div>
          </div>
        </template>
      </div>
      <div class="py-5 grid grid-flow-row gap-3">
        <span class="font-semibold text-heading5">PRODUCT VARIAN GROUP</span>
        <template v-if="payloadVarian.length">
          <div v-for="(payload, i) in payloadVarian" :key="i" class="flex items-center">
            <help-input
              class="w-full"
              type="text"
              v-model="payload.name"
              :bgColor="'gray-100'"
              :disabled="true"
              right-icon="chevron-right"
              :pointer="true"
              @click="handleEditVarian(payload, i)"
            />
          </div>
        </template>
        <template v-else>
          <p class="sm:px-2 sm:ml-2 py-2 text-gold-dark bg-gold-soft rounded-sm text-center inline">
            No variant group found please add some
          </p>
        </template>
        <div class="flex justify-center">
          <span @click="flagVarianGroup = !flagVarianGroup" class="font-medium bg-blue-500 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer hover:bg-blue-400">Set Variation Options</span>
        </div>
      </div>
      <div class="divide-y divide-grey-4">
        <p></p>
        <div class="flex justify-end py-4">
          <span @click="$emit('close')" class="font-medium shadow-sm py-2 px-6 rounded-2xl cursor-pointer border hover:shadow-md">Cancel</span>&nbsp;
          <span @click="$emit('close')" class="font-medium bg-blue-500 shadow-sm py-2 px-6 rounded-2xl text-white cursor-pointer border hover:shadow-md">Save Product</span>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Icon from '@/components/atoms/Icon.vue';
import HelpButton from '@/components/atoms/Button.vue';
import HelpThumbnail from '@/components/atoms/Thumbnail.vue';
import HelpInput from '@/components/atoms/Input.vue';
import HelpSelect from '@/components/molecules/Select.vue';
import HelpModal from '@/components/templates/Modal.vue';
import Confirmation from '@/components/modals/Confirmation.vue';
import VarianOptions from '@/components/sub-components/VarianOptions.vue';
import ProductCatalog from '@/components/sub-components/ProductCatalog.vue';
import mixin from '@/mixin';
import { uuid } from 'uuidv4';
import API from '@/apis';

export default {
  name: 'MerchantItemForm',
  mixins: [mixin],
  components: {
    Icon,
    HelpButton,
    HelpThumbnail,
    HelpInput,
    HelpSelect,
    HelpModal,
    Confirmation,
    VarianOptions,
    ProductCatalog,
  },
  data() {
    return {
      stocks: [
        {
          value: 'AVAILABLE',
          label: 'Tersedia',
        },
        {
          value: 'UNAVAILABLE',
          label: 'Habis',
        },
      ],
      imageFile: null,
      imageIsChanged: false,
      form: {
        src: '',
        productCatalog: 'Pilih Katalog',
        catalog_id: 'Pilih Kategori',
        status: 'Pilih Status',
        varianGroup: {
          name: '',
          required: false,
          multipleChoice: false,
        },
      },
      productImages: [],
      itemCategories: [],
      itemCatalogs: [
        {
          value: 'Paket Nasi',
          label: 'Paket Nasi',
          disabled: true,
          editable: false,
          color: 'grey-7',
        },
        {
          value: 'Paket Minuman',
          label: 'Paket Minuman',
          disabled: true,
          editable: false,
          color: 'grey-7',
        },
        {
          value: 'Paket Dessert',
          label: 'Paket Dessert',
          disabled: true,
          editable: false,
          color: 'grey-7',
        },
        {
          value: 'Paket Beverages',
          label: 'Paket Beverages',
          disabled: true,
          editable: false,
          color: 'grey-7',
        },
      ],
      itemVarians: [
        {
          varianItemName: '',
          additionalPrice: '',
        },
      ],
      payloadVarian: [],
      index: 0,
      photoHover: {},
      modal: {
        sm: false,
      },
      payloadPhoto: '',
      isExistVarian: false,
      flagVarianGroup: false,
      flagEditVarian: false,
      flagItemCatalog: false,
      colorItem: 'gray-100',
      newCatalog: '',
    };
  },
  computed: {
    merchant() {
      return this.$store.state.merchant;
    },
    loading() {
      return this.$store.state.loading.merchant;
    },
    screenWidth() {
      return this.$store.state.screenWidth;
    },
  },
  methods: {
    async generateCatalogs() {
      const {
        data: { data },
      } = await API.get('item-groups');
      this.itemCategories = data.map((el) => ({
        ...el,
        value: el.id,
        label: el.name,
      }));
    },
    async generateVarians() {
      const {
        data: { data },
      } = await API.get('variations');
      console.log(data, 'tes');
    },
    handleChangeImg(e) {
      if (e.target.files.length) {
        const file = e.target.files[0];
        const fileName = `${uuid()}.${e.target.files[0].type.split('/')[1]}`;
        const url = `${this.S3BaseURL}/${fileName}`;
        this.productImages.push(URL.createObjectURL(file));
        this.photoHover[`${this.productImages.length - 1}`] = false;
        this.index += 1;
        this.imageFile = { file, fileName, url };
        this.imageIsChanged = true;
      }
    },
    handleDeletePhoto(payload) {
      this.modal.sm = true;
      this.payloadPhoto = payload;
    },
    handleDeleteCatalog(payload) {
      this.itemCatalogs = this.itemCatalogs.filter((el) => el.value !== payload.value);
    },
    handleAddCatalog() {
      this.itemCatalogs.unshift({
        value: this.newCatalog,
        label: this.newCatalog,
        disabled: true,
        color: 'gray-100',
      });
      this.newCatalog = '';
    },
    handleAddVarian() {
      this.itemVarians.unshift({
        varianItemName: '',
        additionalPrice: '',
      });
    },
    handleEditVarian(payload, i) {
      this.flagVarianGroup = true;
      this.flagEditVarian = true;
      this.form.varianGroup = {
        id: i,
        name: payload.name,
        multipleChoice: payload.multipleChoice,
        required: payload.required,
      };
      this.itemVarians = payload.varians;
    },
    handleResetVarian() {
      if (this.flagEditVarian) {
        this.flagVarianGroup = false;
      } else {
        this.form.varianGroup = {
          name: '',
          required: false,
          multipleChoice: false,
        };
        this.itemVarians = [
          {
            varianItemName: '',
            additionalPrice: '',
          },
        ];
      }
    },
    submitVarian() {
      const {
        id, name, multipleChoice, required,
      } = this.form.varianGroup;
      if (!this.flagEditVarian) {
        this.payloadVarian.push({
          id: this.itemVarians.length - 1,
          name,
          multipleChoice,
          required,
          varians: this.itemVarians,
        });
      } else {
        this.payloadVarian.forEach((el) => {
          console.log(el.id);
          console.log(id);
          if (el.id === id) {
            el.name = name;
            el.multipleChoice = multipleChoice;
            el.required = required;
            el.varians = this.itemVarians;
          }
        });
      }
      this.flagVarianGroup = false;
      this.flagEditVarian = false;
      this.handleResetVarian();
    },
    confirmDelete() {
      if (this.productImages.length === 0) {
        this.productImages = [];
      } else {
        this.productImages = this.productImages.filter((el) => el !== this.payloadPhoto);
      }
      this.modal.sm = false;
    },
    handleHover(index, param) {
      this.photoHover[`${index}`] = param;
    },
    handleEditCatalog(payload, param) {
      payload.label = payload.value;
      if (param === 'edit-name') {
        payload.color = 'gray-100';
        payload.editable = false;
        payload.disabled = true;
      } else {
        payload.color = 'white';
        payload.editable = true;
        payload.disabled = false;
      }
    },
  },
  mounted() {
    this.generateCatalogs();
    this.generateVarians();
  },
};
</script>

<style></style>
