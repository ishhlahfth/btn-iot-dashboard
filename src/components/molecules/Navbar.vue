<template>
  <div
    class="w-full bg-snow flex justify-between items-center p-4 sm:px-6 border-b border-grey-4"
  >
    <div>
      <icon
        name="menu"
        :size="6"
        @click="store.methods.setMini(false)"
        class="block sm:hidden cursor-pointer"
      />
    </div>
    <div class="flex justify-end items-center">
      <p class="font-medium">Name</p>
      <help-avatar
        src="https://www.whiteboardjournal.com/wp-content/uploads/2020/07/Fathia-12.jpg"
        :size="32"
        :tabindex="0"
        class="ml-3 cursor-pointer relative outline-none"
        @click="opened = !opened"
        @blur="opened = false"
      >
        <help-option
          :class="{ hidden: !opened }"
          :options="['Profile', 'Logout']"
          :position="['bottom', 'left']"
          @changeSelected="navigate"
        />
      </help-avatar>
    </div>
  </div>
</template>

<script>
import { ref, inject } from 'vue';
import HelpAvatar from '../atoms/Avatar.vue';
import HelpOption from './Option.vue';
import Icon from '../atoms/Icon.vue';

export default {
  name: 'HelpNavbar',
  components: {
    HelpAvatar,
    HelpOption,
    Icon,
  },
  setup() {
    const store = inject('store');
    const opened = ref(false);
    return {
      store,
      opened,
    };
  },
  methods: {
    navigate(menu) {
      console.log('ðŸ”°', menu);
      if (menu === 'Logout') {
        this.$router.push('/');
      }
    },
  },
};
</script>

<style></style>
