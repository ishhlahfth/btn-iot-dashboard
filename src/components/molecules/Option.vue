<template>
  <div
    class="absolute w-full min-w-max max-h-52 overflow-y-auto p-1 bg-snow rounded-lg border border-grey-4 shadow-lg"
    style="z-index: 999;"
    :class="[
      { 'top-12': position[0] === 'bottom' },
      { 'bottom-12': position[0] === 'top' },
      { 'left-0': position[1] === 'right' },
      { 'right-0': position[1] === 'left' },
    ]"
  >
    <help-option-item
      v-for="(option, i) in options"
      :key="i"
      :label="checkLabel(option)"
      :selected="selected === checkValue(option)"
      @click="$emit('changeSelected', option)"
    />
  </div>
</template>

<script>
import HelpOptionItem from '../atoms/OptionItem.vue';

export default {
  name: 'HelpOption',
  components: {
    HelpOptionItem,
  },
  props: {
    options: {
      type: Array,
      required: true,
    },
    default: {
      type: String,
      default: '',
    },
    selected: {
      type: [String, Number, Object],
      default: '',
    },
    position: {
      type: Array,
      default() {
        return ['bottom', 'right'];
      },
    },
  },
  methods: {
    checkLabel(option) {
      let result = option;
      if (typeof option === 'object') {
        result = option.label;
      }
      return result;
    },
    checkValue(option) {
      let result = option;
      if (typeof option === 'object') {
        result = option.value;
      }
      return result;
    },
  },
};
</script>

<style></style>
