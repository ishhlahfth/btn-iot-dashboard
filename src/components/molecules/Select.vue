<template>
  <div class="relative w-full outline-none" :tabindex="0" @blur="opened = false">
    <div
      class="flex justify-between items-center border border-grey-4 py-2.5 px-3 rounded-lg cursor-pointer select-none"
      :class="{ 'ring-2 ring-royal': opened }"
      @click="opened = !opened"
    >
      <p class="mr-2 truncate">{{ selected }}</p>
      <icon name="selector" />
    </div>
    <help-option
      :class="{ hidden: !opened }"
      :options="options"
      :position="position"
      :selected="modelValue"
      @changeSelected="changeSelected"
    />
  </div>
</template>

<script>
import Icon from '../atoms/Icon.vue';
import HelpOption from './Option.vue';

export default {
  name: 'HelpSelect',
  components: {
    HelpOption,
    Icon,
  },
  props: {
    modelValue: {
      type: [String, Number],
      required: true,
    },
    options: {
      type: Array,
      required: true,
    },
    position: {
      type: Array,
      default() {
        return ['bottom', 'right'];
      },
    },
  },
  data() {
    return {
      selected: !this.modelValue
        ? this.options.length > 0
          ? this.options[0]
          : null
        : this.modelValue,
      opened: false,
    };
  },
  methods: {
    changeSelected(newItem) {
      this.selected = newItem;
      this.opened = false;
      this.$emit('update:modelValue', newItem);
    },
  },
};
</script>

<style lang="scss" scoped></style>
