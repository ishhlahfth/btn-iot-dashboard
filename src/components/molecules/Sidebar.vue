<template>
  <nav
    class="h-full fixed inset-y-0 left-0 sm:grid gap-2 auto-rows-max bg-midnight z-50 sm:p-2 transition-all duration-200"
    :class="store.state.mini ? 'w-0 sm:w-14' : 'p-2 w-64'"
    @mouseover="store.methods.setMini(false)"
    @mouseleave="store.methods.setMini(true)"
  >
    <template v-for="(link, i) in links" :key="i">
      <nav-item v-if="!link.subMenu" :menu="link" />
      <nav-expandable-item
        v-else
        :menu="link"
        ref="expandableItem"
        :miniSidebar="store.state.mini"
      />
    </template>
  </nav>
  <div v-if="!store.state.mini" @click="store.methods.setMini(true)" class="sm:hidden absolute top-0 right-0 bg-black bg-opacity-40 h-full w-full z-20"></div>
</template>

<script>
import { ref, inject } from 'vue';
import NavExpandableItem from '../atoms/NavExpandableItem.vue';
import NavItem from '../atoms/NavItem.vue';

export default {
  name: 'HelpSidebar',
  components: {
    NavExpandableItem,
    NavItem,
  },
  setup() {
    const store = inject('store');
    const collapsed = ref(true);
    const links = [
      { path: '/bns/home', label: 'Home', icon: 'home' },
      { path: '/bns/another-page', label: 'Components', icon: 'user' },
      { path: '/bns/lorem-ipsum', label: 'Templates', icon: 'photograph' },
      {
        path: '/',
        label: 'Dummy Page',
        icon: 'cube',
        subMenu: [
          { path: '/bns/dummy-page-1', label: 'First', icon: '' },
          { path: '/bns/dummy-page-2', label: 'Second', icon: '' },
          { path: '/bns/dummy-page-3', label: 'Third', icon: '' },
        ],
      },
    ];
    return {
      store,
      collapsed,
      links,
    };
  },
};
</script>

<style lang="scss" scoped>
.slide-fade-enter-active,
.slide-fade-leave-active {
  transition: all 0.2s ease;
}
.slide-fade-enter-from,
.slide-fade-leave-to {
  transform: translateY(-10px);
  opacity: 0;
}
.rotate {
  transform: rotate(180deg);
}
// nav {
//   &::before {
//     content: '';
//     position: absolute;
//     right: -50%;
//     top: 0;
//     height: 100vh;
//     width: 100%;
//     background: rgba($color: #000000, $alpha: 0.4);
//   }
// }
</style>
