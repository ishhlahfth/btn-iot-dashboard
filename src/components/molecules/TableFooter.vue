<template>
  <tfoot class="w-full px-6 py-2 flex items-center justify-between bg-white">
    <p class="text-small">
      Showing
      <span class="font-medium">{{ firstRow }}</span>
      to
      <span class="font-medium">{{ lastRow }}</span>
      of
      <span class="font-medium">{{ totalRows }}</span>
      results
    </p>

    <div class="flex items-center text-small">
      <span class="mr-2">Records per page</span>
      <div class="mr-2">
        <help-select
          :options="[10, 25, 50, 100]"
          :position="['top', 'left']"
          v-model="localRowLimit"
        />
      </div>
      <div class="flex bg-white rounded-md divide-x-2 border border-grey-4">
        <div
          class="p-2 rounded-l-md"
          :class="localPage <= 1 ? 'bg-grey-5 cursor-not-allowed' : 'cursor-pointer'"
          @click="previousPage"
        >
          <svg
            class="w-5 h-5"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 20 20"
            :fill="localPage <= 1 ? '#C4C4C4' : 'currentColor'"
          >
            <path
              fillRule="evenodd"
              d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
              clipRule="evenodd"
            />
          </svg>
        </div>
        <div
          class="p-2 rounded-r-md"
          :class="lastRow === totalRows ? 'bg-grey-5 cursor-not-allowed' : 'cursor-pointer'"
          @click="nextPage"
        >
          <svg
            class="w-5 h-5"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 20 20"
            :fill="lastRow === totalRows ? '#C4C4C4' : 'currentColor'"
          >
            <path
              fillRule="evenodd"
              d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
              clipRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </tfoot>
</template>

<script>
import HelpSelect from './Select.vue';

export default {
  name: 'TableFooter',
  components: {
    HelpSelect,
  },
  props: {
    page: {
      type: Number,
      required: true,
    },
    rowLimit: {
      type: Number,
      required: true,
    },
    totalRows: {
      type: Number,
      required: true,
    },
  },
  data() {
    return {
      localRowLimit: 10,
      localPage: 1,
    };
  },
  computed: {
    firstRow() {
      let firstRow = 0;
      if (this.totalRows !== 0) {
        firstRow = this.page * this.rowLimit - this.rowLimit + 1;
      }
      return firstRow;
    },
    lastRow() {
      let lastRow = 0;
      const ceiledRow = this.page * this.rowLimit;
      if (ceiledRow > this.totalRows) {
        lastRow = this.totalRows;
      } else {
        lastRow = ceiledRow;
      }
      return lastRow;
    },
  },
  watch: {
    localRowLimit(newValue) {
      this.$emit('onChangePagination', {
        rowLimit: newValue,
        page: 1,
      });
      this.localPage = 1;
    },
    localPage(newValue) {
      this.$emit('onChangePagination', {
        rowLimit: this.localRowLimit,
        page: newValue,
      });
    },
  },
  methods: {
    nextPage() {
      if (this.lastRow !== this.totalRows) this.localPage += 1;
    },
    previousPage() {
      if (this.localPage > 0) this.localPage -= 1;
    },
  },
  mounted() {
    this.localRowLimit = this.rowLimit;
    this.localPage = this.page;
  },
};
</script>

<style></style>
