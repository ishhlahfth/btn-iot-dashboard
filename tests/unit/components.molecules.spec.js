import MenuCard from '@/components/molecules/MenuCard.vue';
import MerchantItemForm from '@/components/molecules/MerchantItemForm.vue';
import Navbar from '@/components/molecules/Navbar.vue';
import SummaryCard from '@/components/molecules/SummaryCard.vue';
import Option from '@/components/molecules/Option.vue';
import Select from '@/components/molecules/Select.vue';
import Sidebar from '@/components/molecules/Sidebar.vue';
import TableFooter from '@/components/molecules/TableFooter.vue';

// import { mount, shallowMount } from '@vue/test-utils';

describe('MenuCard.vue', () => {
  it('should render correctly', () => {
    expect(MenuCard.$el).toMatchSnapshot();
  });

  it('should render name MenuCard', () => {
    expect(MenuCard.name).toBe('MenuCard');
  });

  // it('should render all props', () => {
  //   const wrapper = mount(MenuCard, {
  //     props: {
  //       raw: {
  //         name: 'test'
  //       },
  //       imageUrl: 'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png',
  //       name: 'Untitled',
  //       category: '-',
  //       description: '-',
  //       price: 0,
  //       availabilityStatus: 'tersedia',
  //       isActive: true,
  //       variants: [],
  //       loading: false,
  //       isForProduct: true,
  //       isEdit: true,
  //     },
  //   });
  //   expect(wrapper.props().isEdit).toBeTruthy();
  // });
});

describe('MerchantItemForm.vue', () => {
  it('should render correctly', () => {
    expect(MerchantItemForm.$el).toMatchSnapshot();
  });

  it('should render name MerchantItemForm', () => {
    expect(MerchantItemForm.name).toBe('MerchantItemForm');
  });
});

describe('Navbar.vue', () => {
  it('should render correctly', () => {
    expect(Navbar.$el).toMatchSnapshot();
  });

  it('should render name Navbar', () => {
    expect(Navbar.name).toBe('HelpNavbar');
  });
});

describe('SummaryCard.vue', () => {
  it('should render correctly', () => {
    expect(SummaryCard.$el).toMatchSnapshot();
  });

  it('should render name SummaryCard', () => {
    expect(SummaryCard.name).toBe('SummaryCard');
  });
});

describe('Option.vue', () => {
  it('should render correctly', () => {
    expect(Option.$el).toMatchSnapshot();
  });

  it('should render name Option', () => {
    expect(Option.name).toBe('HelpOption');
  });
});

describe('Select.vue', () => {
  it('should render correctly', () => {
    expect(Select.$el).toMatchSnapshot();
  });

  it('should render name Select', () => {
    expect(Select.name).toBe('HelpSelect');
  });
});

describe('Sidebar.vue', () => {
  it('should render correctly', () => {
    expect(Sidebar.$el).toMatchSnapshot();
  });

  it('should render name Sidebar', () => {
    expect(Sidebar.name).toBe('HelpSidebar');
  });
});

describe('TableFooter.vue', () => {
  it('should render correctly', () => {
    expect(TableFooter.$el).toMatchSnapshot();
  });

  it('should render name TableFooter', () => {
    expect(TableFooter.name).toBe('TableFooter');
  });
});

