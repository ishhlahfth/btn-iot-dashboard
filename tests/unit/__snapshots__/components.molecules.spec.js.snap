// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCard.vue should render correctly 1`] = `undefined`;

exports[`MerchantItemForm.vue should render correctly 1`] = `undefined`;

exports[`Navbar.vue should render correctly 1`] = `undefined`;

exports[`Option.vue should render correctly 1`] = `undefined`;

exports[`Select.vue should render correctly 1`] = `undefined`;

exports[`Sidebar.vue should render correctly 1`] = `undefined`;

exports[`SummaryCard.vue should render correctly 1`] = `undefined`;

exports[`TableFooter.vue should render correctly 1`] = `undefined`;
